vm/object.c

```diff
@@ -64,27 +64,41 @@ int vm_objectGet(vm_object_t **o, oid_t oid)
 	*o = lib_treeof(vm_object_t, linkage, lib_rbFind(&object_common.tree, &t.linkage));
 
 	if (*o == NULL) {
+		/* Take off the lock to avoid a deadlock in vm_kmalloc */
+		proc_lockClear(&object_common.lock);
+
 		sz = (size_t)proc_size(oid);
 		n = round_page(sz) / SIZE_PAGE;
 
-		if ((*o = (vm_object_t *)vm_kmalloc(sizeof(vm_object_t) + n * sizeof(page_t *))) == NULL) {
-			proc_lockClear(&object_common.lock);
+		if ((no = (vm_object_t *)vm_kmalloc(sizeof(vm_object_t) + n * sizeof(page_t *))) == NULL) {
```

Comment:
```suggestion
		no = (vm_object_t *)vm_kmalloc(sizeof(vm_object_t) + n * sizeof(page_t *));
		if (no == NULL) {
```
What happens if `proc_size()` returns a negative value e.g. `-ENOMEM`.